include/rdma/fi_domain.h

```diff
@@ -112,6 +112,11 @@ struct fid_mr {
 	uint64_t		key;
 };
 
+enum fi_hmem_iface {
+	FI_HMEM_NONE	= 0,
```

Comment:
`FI_HMEM_HOST` maybe, so the default attribute for host buffer registrations can be of that type?

Reply:
Either is fine to me. The "NONE" version emphasizes that no device calls are involved with the management of the memory. The "HOST" version emphasizes the type of the memory.

Reply:
Yeah, I was more leaning toward the type of memory for these HMEM changes and not make this attribute relevant only if a peer device is used.

Reply:
My intent with these was to emphasize what APIs were used to allocate the memory, rather than necessarily indicate where the memory came from.  The 0 value indicates that the app called something like malloc, calloc, mmap, etc..  I.e. no special APIs were used in the allocation.  Even CUDA allows allocating host memory.

Reply:
I see your point. So a host buffer allocated with CUDA would be marked FI_HMEM_CUDA even if it is a host buffer. Makes sense.